### PR TITLE
GSdx: Remove some broken buffer width correction code

### LINKED ISSUE #3982: [Shifted Graphics] Galaxy Angel
**Describe the bug**
graphics are shifted in game

**To Reproduce**
play game. 

**Expected behavior**

**GS Settings**
does it on both hardware and software renderer

**GS Window Screenshots**
gsdx_20201207005807

**System Info (please complete the following information):**
 - PCSX2 Revision: 1.7.0-dev-670
 - OS: Windows 10
 - CPU: i5-4430
 - GPU: GTX 960

**Logs and Dumps**
[gsdx_20201207005807.gs.zip](https://github.com/PCSX2/pcsx2/files/5650968/gsdx_20201207005807.gs.zip)

### PR DESCRIPTION
**Pull request by tadanokojin**
Removes some code that was meant to correct a condition on the GS where 8-bit and 4-bit formats are assigned a buffer width that is not divisible by 2. Given that the code just forced it to be even (almost certainly not correct) and was some kind of workaround dating back 13 years I'm opting to remove it. If someone can give me an example of a game that is broken without this code I'll look into it again but it seems like a fairly rare condition. The examples I could find aren't particularly interesting:

- namcoXCapcom doesn't appear to have any problems absent this code and no one documented the problem to begin with so it's near impossible to track down. If anyone can point me to an issue in this game I'd appreciate that.
- Geomon is clearly broken regardless of this code and spams a bunch of TEX0 writes with complete nonsense values. Either way I didn't see an issue in that game either in so far as this would be the problem.
- My tests did not produce any real difference

As a bonus I made GSState a little more pretty.

fixes #3982

### CODE REVIEW

File: plugins/GSdx/GSState.cpp [hunk 1]
```diff
@@ -153,7 +154,7 @@ GSState::GSState()
 	m_sssize += sizeof(m_q);
 
 	PRIM = &m_env.PRIM;
-//	CSR->rREV = 0x20;
+	//	CSR->rREV = 0x20;
```
Comment (refractionpcsx2) [inline plugins/GSdx/GSState.cpp:157]: Do you wanna tidy that up, that looks weird now. Thanks Clang :P

File: plugins/GSdx/GSState.cpp [hunk 2]
```diff
@@ -237,8 +241,8 @@ void GSState::Reset()
 	memset(&m_path[0], 0, sizeof(m_path[0]) * countof(m_path));
 	memset(&m_v, 0, sizeof(m_v));
 
-//	PRIM = &m_env.PRIM;
-//	m_env.PRMODECONT.AC = 1;
+	//	PRIM = &m_env.PRIM;
+	//	m_env.PRMODECONT.AC = 1;
```
Comment (refractionpcsx2) [inline plugins/GSdx/GSState.cpp:245]: Tidy these too

File: plugins/GSdx/GSState.cpp [hunk 3]
```diff
@@ -658,10 +672,10 @@ void GSState::GIFPackedRegHandlerXYZF2(const GIFPackedReg* RESTRICT r)
 	VertexKick<prim, auto_flush>(adc ? 1 : r->XYZF2.Skip());
 }
 
-template<uint32 prim, uint32 adc, bool auto_flush>
+template <uint32 prim, uint32 adc, bool auto_flush>
 void GSState::GIFPackedRegHandlerXYZ2(const GIFPackedReg* RESTRICT r)
 {
-/*
+	/*
```
Comment (refractionpcsx2) [inline plugins/GSdx/GSState.cpp:678]: weird, it indented this one but not the closing comment tag

File: plugins/GSdx/GSState.cpp [hunk 4]
```diff
@@ -859,17 +873,17 @@ void GSState::GIFRegHandlerUV_Hack(const GIFReg* RESTRICT r)
 	m_isPackedUV_HackFlag = false;
 }
 
-template<uint32 prim, uint32 adc, bool auto_flush>
+template <uint32 prim, uint32 adc, bool auto_flush>
 void GSState::GIFRegHandlerXYZF2(const GIFReg* RESTRICT r)
 {
-/*
+	/*
```
Comment (refractionpcsx2) [inline plugins/GSdx/GSState.cpp:879]: and this one

File: plugins/GSdx/GSState.cpp [hunk 5]
```diff
@@ -859,17 +873,17 @@ void GSState::GIFRegHandlerUV_Hack(const GIFReg* RESTRICT r)
...
 	m_v.XYZ.X = r->XYZF.X;
 	m_v.XYZ.Y = r->XYZF.Y;
 	m_v.XYZ.Z = r->XYZF.Z;
 	m_v.FOG.F = r->XYZF.F;
 */
-	
-/*
+
+	/*
```
Comment (refractionpcsx2) [inline plugins/GSdx/GSState.cpp:886]: And this one

File: plugins/GSdx/GSState.cpp [hunk 6]
```diff
@@ -1554,7 +1596,7 @@ void GSState::FlushWrite()
 
 	InvalidateVideoMem(m_env.BITBLTBUF, r);
 	*/
-/*
+	/*
```
Comment (refractionpcsx2) [inline plugins/GSdx/GSState.cpp:1599]: more clang indenting weirdness not matching the closing tag

File: plugins/GSdx/GSState.cpp [hunk 7]
```diff
@@ -1817,30 +1876,32 @@ void GSState::Read(uint8* mem, int len)
...
 	m_mem.ReadImageX(m_tr.x, m_tr.y, mem, len, m_env.BITBLTBUF, m_env.TRXPOS, m_env.TRXREG);
 
-	if(s_dump && s_save && s_n >= s_saven) {
+	if (s_dump && s_save && s_n >= s_saven)
+	{
 		std::string s = m_dump_root + format("%05d_read_%05x_%d_%d_%d_%d_%d_%d.bmp",
-				s_n, (int)m_env.BITBLTBUF.SBP, (int)m_env.BITBLTBUF.SBW, (int)m_env.BITBLTBUF.SPSM,
-				r.left, r.top, r.right, r.bottom);
+											 s_n, (int)m_env.BITBLTBUF.SBP, (int)m_env.BITBLTBUF.SBW, (int)m_env.BITBLTBUF.SPSM,
+											 r.left, r.top, r.right, r.bottom);
```
Comment (refractionpcsx2) [inline plugins/GSdx/GSState.cpp:1904]: Dunno if you wanna un-indent this or not, just looks a bit weird like this out so far

File: plugins/GSdx/GSState.cpp [hunk 8]
```diff
@@ -1858,27 +1919,37 @@ void GSState::Move()
...
+		dx += w - 1;
+		xinc = -1;
+	}
+	if (m_env.TRXPOS.DIRY)
+	{
+		sy += h - 1;
+		dy += h - 1;
+		yinc = -1;
+	}
+	/*
```
Comment (refractionpcsx2) [inline plugins/GSdx/GSState.cpp:1945]: more clang indenting weirdness

File: plugins/GSdx/GSState.cpp [hunk 9]
```diff
@@ -1858,27 +1919,37 @@ void GSState::Move()
...
 	printf("%05x %d %d => %05x %d %d (%d%d), %d %d %d %d %d %d\n",
 		m_env.BITBLTBUF.SBP, m_env.BITBLTBUF.SBW, m_env.BITBLTBUF.SPSM,
 		m_env.BITBLTBUF.DBP, m_env.BITBLTBUF.DBW, m_env.BITBLTBUF.DPSM,
 		m_env.TRXPOS.DIRX, m_env.TRXPOS.DIRY,
 		sx, sy, dx, dy, w, h);
 */
-/*
+	/*
```
Comment (refractionpcsx2) [inline plugins/GSdx/GSState.cpp:1952]: more clang indenting weirdness

**Review by refractionpcsx2 — COMMENTED** [review]
Just a few tidyness changes should you wish to tidy :)

File: plugins/GSdx/GSState.cpp [hunk 10]
```diff
@@ -894,8 +908,22 @@ void GSState::GIFRegHandlerXYZ2(const GIFReg* RESTRICT r)
...
 }
 
-template<int i> void GSState::ApplyTEX0(GIFRegTEX0& TEX0)
+template <int i>
+void GSState::ApplyTEX0(GIFRegTEX0& TEX0)
 {
+	// TODO: Paletted Formats
+	// 8-bit and 4 bit formats need to be addressed with a buffer width divisible 2.
+	// However, not doing so is possible and does have a bahavior on the GS.
+	// When immplementing such code care must be taken not to apply it unless it is
```
Comment (RedDevilus) [inline plugins/GSdx/GSState.cpp:917]: Typo: immplementing, bahavior

**Review by refractionpcsx2 — APPROVED** [review]

**Comment by seta-san** [comment]
gsdx_20201209020405
gsdx_20201209020310

**Pull request merged**